### ISSUE TITLE
Unavailable proxy with udp broke all connections 

### ISSUE THREAD

**Issue by psydvl** [AdguardTeam/AdguardForAndroid#4808]
### Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

4.1 Nightly 5

### Environment

```markdown
- OS: Android 13
- Device: Oneplus 11
- Firmware: (what is it?)
```

### Root access

- [ ] Yes, I have it.

### Issue Details

Steps to reproduce:
1. Add any not working proxy (for example 127.0.0.1:11)
2. Enable udp routing for this proxy 
3. Enable proxy 

### Expected Behavior

All apps not routing through proxy keep working

### Actual Behavior

Almost all apps stop working 

### Screenshots

<details><summary>Screenshot proxy:</summary>

![Screenshot_2023-06-19-11-02-02-51](https://github.com/AdguardTeam/AdguardForAndroid/assets/43755002/56fecb52-5960-4fcc-824b-80c26a168381)

</details>
<details><summary>Screenshot Firefox:</summary>

![Screenshot_2023-06-19-11-02-33-47_3aea4af51f236e4932235fdada7d1643](https://github.com/AdguardTeam/AdguardForAndroid/assets/43755002/988bde92-e516-4674-957a-382e5b9b8cd8)

</details>

### Additional Information

Probably related to https://github.com/AdguardTeam/AdguardForAndroid/issues/4769

**Comment by Versty** [comment]
@psydvl Thank you for reporting. 
This issue reproduces on our side as well.

**Comment by Versty** [comment]
@psydvl This bug should be fixed in the AdGuard for Android 4.0 nightly 7. 
Could you check if it is gone on your device?

**Comment by psydvl** [comment]
Yes, solved for me
